File: cli/src/modules/compose/redis.ts
import type { ComposeService } from '@/core/types.js';

export const redisService: ComposeService = {
  id: 'redis',
  label: 'Redis',
  volumes: ['redis_data'],
  options: [
    {
      id: 'version',
      label: 'Redis version',
      type: 'select',
      choices: [
        { value: '7.4-alpine', label: 'Redis 7.4 (BSD-licensed last)' },
        { value: '8.6-alpine', label: 'Redis 8.6 (latest, SSPL/RSALv2)' },
        { value: '8.0-alpine', label: 'Redis 8.0' },
      ],
      default: '7.4-alpine',
    },
  ],
  render({ options }) {
    const version = (options.version as string) || '7.4-alpine';
    return {
      image: `redis:${version}`,
      container_name: 'redis',
      volumes: ['redis_data:/data'],
      networks: ['local-network'],
    };
  },
};

File: cli-go/internal/modules/compose/redis.go
package compose

import (
	"fmt"

	"github.com/joacohbc/my-devcontainer-installer/cli-go/internal/core"
)

var RedisService = &ServiceSpec{
	ID:      "redis",
	Label:   "Redis",
	Volumes: []string{"redis_data"},
	Options: []core.ModuleOption{
		{
			ID:    "version",
			Label: "Redis version",
			Type:  core.ModuleOptionSelect,
			Choices: []core.ModuleOptionChoice{
				{Value: "7.4-alpine", Label: "Redis 7.4 (BSD-licensed last)"},
				{Value: "8.6-alpine", Label: "Redis 8.6 (latest, SSPL/RSALv2)"},
				{Value: "8.0-alpine", Label: "Redis 8.0"},
			},
			Default: "7.4-alpine",
		},
	},
	Render: func(ctx RenderContext) *ServiceDef {
		version, _ := ctx.Options["version"].(string)
		if version == "" {
			version = "7.4-alpine"
		}
		return &ServiceDef{
			Image:         fmt.Sprintf("redis:%s", version),
			ContainerName: "redis",
			Volumes:       []string{"redis_data:/data"},
			Networks:      []string{"local-network"},
		}
	},
}
